scipy/optimize/_bracket.py

```diff
@@ -45,6 +52,11 @@ def _bracket_root_iv(func, xl0, xr0, xmin, xmax, factor, args, maxiter):
     if not xp.all(factor > 1):
         raise ValueError('All elements of `factor` must be greater than 1.')
 
+    # Calculate the default value of xr0 if a value has not been supplied.
+    # Be careful to ensure xr0 is not larger than xmax.
+    if xr0_not_supplied:
+        xr0 = xl0 + xp.minimum((xmax - xl0)/ 8, xp.asarray(1.0))
```

Comment:
Looks like this would not preserve lower-precision dtypes in libraries for which `float64` is the default.
```suggestion
        xr0 = xl0 + xp.minimum((xmax - xl0)/ 8, xp.asarray(1.0, dtype=xl0.dtype))
```
Could you confirm locally?

Eagerly waiting the next version of the standard, which will accept Python scalars where it makes sense.

Reply:
Yep, I noticed independently and pushed a fix just now. The same bug was already there in `bracket_minimum` so I fixed it too.